de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/views/SarosView.java

```diff
@@ -441,6 +446,13 @@ public void menuAboutToShow(final IMenuManager manager) {
              * version 14.1.31)
              */
             // manager.add(getAction(SkypeAction.class));
+
+            // TODO: Replace Server check once other implementations support JoinSessionRequests
+            if (contacts.size() == 1
+                && discoveryManager.queryFeatureSupport(
```

Comment:
Beside the NPE that can be thrown this can cause UI freezes, so I am sorry this is not something that is suiteable for production.

Reply:
Well, it would allow for an alpha release of the saros server (for testing).

I could hide this behind an experimental feature flag, if you like, but as it only activates in the context of using a very experimental server build, I do not really see the point.

That said the whole server topic is really new and I get if at this point this is nothing to be merged. I will then just keep this PR for reference and demonstration purposes.

Reply:
If this is just about the `discoveryManager` call, what about adding an additional function the `discoveryManager` that only checks the cache in therefor is not blocking?

Reply:
The DiscoveryManager has 3 public methods, should not be to hard the one you need: https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/net/xmpp/discovery/DiscoveryManager.java#L202

Reply:
Done